aten/src/ATen/native/cuda/BinaryOpsKernel.cu

```diff
@@ -26,7 +26,7 @@ static void sub_kernel_cuda(TensorIterator& iter, Scalar alpha_scalar) {
 }
 
 void div_kernel_cuda(TensorIterator& iter) {
-  if (!isIntegralType(iter.dtype()) && iter.is_cpu_scalar(2)) {
+  if (!isIntegralType(iter.dtype(), /*includeBool*/ false) && iter.is_cpu_scalar(2)) {
```

Comment:
div for bools wasn't enabled before and this change shouldn't enable it now

Reply:
this looks wrong, because it could enter this for loop which is trying to do floating-point math

Reply:
AT_DISPATCH_FLOATING_TYPES_AND_HALF doenst include bool type so you will get an error like "not implemented for CUDA Bool..." as before

Reply:
ok, but that's different than your original comment which is about actually enabling bool.